cli/src/text_util.rs

```diff
@@ -35,6 +37,33 @@ pub fn split_email(email: &str) -> (&str, Option<&str>) {
     }
 }
 
+pub fn render_copied_path(source: &str, target: &str) -> String {
+    if source == target {
+        return target.into();
+    }
+    let prefix = UnicodeSegmentation::split_word_bound_indices(source)
+        .zip(UnicodeSegmentation::split_word_bound_indices(target))
+        .take_while_inclusive(|((_, source_word), (_, target_word))| source_word == target_word)
+        .last()
+        .map(|((source_index, _), (target_index, _))| (source_index, target_index))
+        .unwrap_or((0, 0));
+    let suffix = UnicodeSegmentation::split_word_bound_indices(source)
+        .rev()
+        .zip(UnicodeSegmentation::split_word_bound_indices(target).rev())
+        .take_while(|((_, source_word), (_, target_word))| source_word == target_word)
+        .last()
+        .map(|((source_index, _), (target_index, _))| (source_index, target_index))
+        .unwrap_or((source.len(), target.len()));
```

Comment:
Is it intentional that `file-1 => file-2` is rendered as `file-{1 => 2}`? I personally find it's hard to follow, and would rather want to split by path separator, not at word boundary.

BTW, a generic implementation of this function would be something like
```rust
Diff::for_tokenizer([source, target], find_path_component_ranges)
    .map(|hunk| match hunk {
        DiffHunk::Matching(common) => common,
        DiffHunk::Different([source, target]) => "{source} => {target}", // XXX handle trailing sep?
     })
     .collect()
```

Reply:
That is a good point.  I have moved this into `repo_path` so it sits with the other format function for changing paths into UI strings.  I have also update the implementation to be component based (which means we no long have a dep on unicode).